patient-care/webapp/server/methods.js

```diff
@@ -125,16 +140,37 @@ Meteor.methods({
         Meteor.call("getSampleGroupVersion", sampleGroup.name);
 
     // ensure uniqueness for data sets
-    let uniqueDataSets = _.uniq(_.pluck(sampleGroup.data_sets, "data_set_id"));
-    if (uniqueDataSets.length !== sampleGroup.data_sets.length) {
+    let dataSetIds = _.uniq(_.pluck(sampleGroup.data_sets, "data_set_id"));
+    if (dataSetIds.length !== sampleGroup.data_sets.length) {
       throw new Meteor.Error("non-unique-data-sets");
     }
 
+    // keep track of each sample label, throw an error if a label exists in
+    // multiple data sets
+    let sampleLabelIndex = {};
+
+    // Store each data sets' feature labels list to the hash map
+    // organized by data set _id. This is used at the end to compute
+    // the intersection of the feature labels.
+    let dataSetFeaturesHash = {};
+
+    // utility function for adding to the dataSetFeaturesHash
+    function addToDSFeatureLabelHash(dataSetId, sampleLabels) {
```

Comment:
I meant to name `sampleLabels` `featureLabels`. This is a copy-paste error and is removed in the latest of what I'm working on on `sg-from-sg`.